File: app/src/main/java/com/anokmik/tripassistant/author/profile/ProfileContract.java
package com.anokmik.tripassistant.author.profile;

public interface ProfileContract {

    interface View {

        void setProfileTitle();

        void setEditAuthorTitle();

        void back();

    }

    interface Presenter {

        void save();

        void cancel();

        void edit();

        void delete();

        void signOut();

    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/author/profile/ProfileContract.kt
package com.anokmik.tripassistant.author.profile

interface ProfileContract {

    interface View {

        fun setProfileTitle()

        fun setEditAuthorTitle()

        fun back()

    }

    interface Presenter {

        fun save()

        fun cancel()

        fun edit()

        fun delete()

        fun signOut()

    }

}